crates/payload/basic/src/builder_stack.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{
+    BuildArguments, BuildOutcome, MissingPayloadBehaviour, PayloadBuilder, PayloadBuilderError,
+    PayloadConfig,
+};
+
+/// A stack of payload builders that can be used in sequence.
+#[derive(Clone, Debug)]
+pub struct PayloadBuilderStack<B>(Vec<B>);
```

Comment:
take a look at this type for reference:

https://docs.rs/tower/latest/tower/layer/util/struct.Stack.html

because this has two instances, this allows "stacking" of multiple different types, which can be nested and is very flexible, for example: `Stack<Stack<A, B>, C>` etc.

A vec would work if all payloadtypes are the same but ideally we can be a bit more flexible here by using two generics

Reply:
Ok I understand now, will fix soon!